runtime/ts/schema.ts

```diff
@@ -390,16 +391,20 @@ export class Schema {
         }
       });
     }
-    return clazz;
+    // TODO: this type-erases the dynamically generated clazz so we
+    // can force it into an Entity type.
+    // tslint:disable-next-line: no-any
+    const c: any = clazz;
+    return c as typeof Entity;
```

Comment:
doesn't clazz extend Entity?

Reply:
Note: it appears that what I just said was incorrect, since this code compiles and generates d.ts files correctly without type erasure.